main.tf

```diff
@@ -17,21 +17,30 @@ resource "aws_security_group" "default" {
   vpc_id = var.vpc_id
   name   = module.label.id
 
-  ingress {
-    from_port       = var.port # Redis
-    to_port         = var.port
-    protocol        = "tcp"
-    security_groups = var.security_groups
-  }
+  tags = module.label.tags
+}
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+resource "aws_security_group_rule" "default_ingress" {
+  for_each                 = toset(var.security_groups)
```

Comment:
we need to be able to enable/disable `aws_security_group_rule` since the whole module supports this contract.

count  = var.enabled ? 1 : 0

but since you added `for_each`, we can do something like this:

for_each                 = var.enabled ? toset(var.security_groups) : []